### PR TITLE
[MacOS][Build][OpenMP] Add OpenMP_ROOT to CMakeLists 

### LINKED ISSUE #1355: [MacOS] [OpenMP] openmp include path should be in default cxx flags
**Describe the motivation for the feature request**

Hello, 
as in the MacOS CI, the include path to openMP must be passed down as a CXX flags when compiling with acpp. It would be better if the such include instead would be in the config for romp cxx flags.

https://github.com/AdaptiveCpp/AdaptiveCpp/blob/c8d5224bf994aee577a02dc513c6563333021005/.github/workflows/macos.yml#L36

**Describe the solution you'd like**

on MacOS by default add the openmp (the outputs of `echo -I$OMP_ROOT/include` in the CI would do) path to `default-omp-cxx-flags`

**Additional context**

For newcomers it make adoption easier on MacOS

### PR DESCRIPTION
**Pull request by badumbatish**
This helps create an easier and more robust build for beginners, fixes #1355

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -339,6 +339,10 @@ endif()
 
 if(APPLE)
   set(DEFAULT_OMP_FLAG "-Xclang -fopenmp")
+  execute_process(COMMAND brew --prefix
```
Comment (fodinabor) [inline CMakeLists.txt]: This won't allow for anyone to build OpenMP themselves...
So maybe only do all of this if `OpenMP_ROOT` is undefined?

Also, I'm not much of a Mac person, so I'm not sure if it makes a difference - in the CI we do the following to identify the full `OpenMP_ROOT` - is `prefix/opt/libomp` always going to be the correct path?
https://github.com/AdaptiveCpp/AdaptiveCpp/blob/0359cac9e08d44c48c586ef3ec88e95ac64a4d44/.github/workflows/macos.yml#L23

**Review by fodinabor — COMMENTED** [review]
Thanks for looking into this!

File: CMakeLists.txt (hunk 1 above)
Comment (badumbatish) [reply CMakeLists.txt]: Hi there, thanks for the review, I haven't looked into the yml yesterday but I've modified the `execute_process`  cmake command to have the same effect as `brew list libomp | grep libomp.a | sed -E "s/\/lib\/.*//"`.

Also add an IF(NOT DEFINED OpenMP_ROOT) to make sure other people can build it

Comment (fodinabor) [reply CMakeLists.txt]: Hi, thanks!
I just realized that still will override manually specified hints such as the environment variable `OMP_ROOT`.
So, maybe it makes sense to move this logic to where we're actually looking for OpenMP and first doing an `find_package(OpenMP QUIET)`, and if that fails, add the `OpenMP_ROOT` and search with `REQUIRED`.

The relevant location will be: https://github.com/AdaptiveCpp/AdaptiveCpp/blob/0359cac9e08d44c48c586ef3ec88e95ac64a4d44/src/runtime/CMakeLists.txt#L280

To show this working in CI, maybe remove the line I referenced earlier from the Action yml... The CI should the still work with your changes.

Comment (badumbatish) [reply CMakeLists.txt]: hmm i think on MacOS, CMake can only find OpenMP if either OMP_ROOT or OpenMP_ROOT is defined. When I did what you suggested first with find_package(OpenMP QUIET) before searching for OpenMP_ROOT, CMake can't seem to find the OpenMP package.

I also removed the old checks in CMakelists and remove the OMP_ROOT on yml

**Comment by badumbatish** [comment]
It seems that automation of build/ is good but failed on actual linkage, maybe we didn't actually define an include path that is based on OpenMP_ROOT inside of CMakeLists.txt, do you want me to add that directly to CXXFlags just as the yml is doing?

**Comment by badumbatish** [comment]
hmm i see the problem now, I don't think the CMakeLists.txt in tests/ gets any information about the OpenMP_ROOT variable in CMakeLists.txt in runtime/

**Comment by badumbatish** [comment]
This should run fine now on github actions, I just want to make sure there's a way to actually use another variables like OPENMP_MACOS_INCLUDE_DIR instead of coupling the thing into CMakeLists.txt just like the yml.

I've refactored the appending of CMAKE_CXX_FLAGS into different target_include_dir of different targets in tests/ now.

I'd love some feedbacks @fodinabor

**Comment by illuhad** [comment]
@fodinabor Do you still have some concerns?

**Review by fodinabor — APPROVED** [review]
Looks good now!
Thanks, and sorry for not responding earlier, didn't really monitor Github due to too many other things happening.

**Comment by badumbatish** [comment]
No worries! I'm just happy every build passes now :)

**Pull request merged**